### PR TITLE
Add permissions filter on the relation field

### PR DESCRIPTION
**Pull request by lauriejim**
### What does it do?

Based on that issue: https://github.com/strapi/strapi/issues/18889

When you have a relation field, and you don't have permission to READ an entry (based on the permissions, isCreator for example), 
the relations will not be displayed in the dropdown list.

### Why is it needed?

It's necessary to respect the READ permission the role has.

### How to test it?

I didn't write any tests for that, so please provide some thoughts on how to test all the cases.
To me I think that everything works fine on what I manually tested.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/18889

### CODE REVIEW

**Comment by joshuaellis** [comment]
This needs to be aimed at develop

**Comment by Boegie19** [comment]
Could this not also cause some Editor Experience issues. since you can click the show more button multiple times without anyting apearing.

File: packages/core/content-manager/server/controllers/relations.js [hunk 1]
```diff
@@ -55,6 +55,14 @@ module.exports = {
...
     await validateFindAvailable(ctx.request.query);
 
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.read()) {
+      return ctx.forbidden();
+    }
+
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(ctx.request.query);
+
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/controllers/relations.js:65]: Uppon further inspection I realized our desired behaviour when the user doesnt have permissions to see a relation is not to completely hide it, but rather hide the main field.

At the moment this was working for normal permissions but not for conditional ones. (You can see the behaviour in the `sanitizeMainField` function above.

Let me give it a thought this week! And see if we can come with another solution, or we commit to your approach (which will need product aproval)

Thank you for the contribution Jim, glad to see you contributing 🚀 
Give me another day and we will come with an answer 🙏 

cc @innerdvations

**Comment by maZahaca** [comment]
Any luck to move this further?

File: packages/core/content-manager/server/controllers/relations.js (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/content-manager/server/controllers/relations.js:65]: agree, I am waiting for a product team response ( lots of holidays during december 🙏 ).

**Comment by RiteshSaxena** [comment]
`const permissionChecker = getService('permission-checker').create({
      userAbility,
      model: attribute.target
    });`
    
 Permission checker needs to be targetting the model of the target field, instead of the current model

**Comment by RiteshSaxena** [comment]
> `const permissionChecker = getService('permission-checker').create({ userAbility, model: attribute.target });`
> 
> Permission checker needs to be targetting the model of the target field, instead of the current model

I have opened a new PR with the updated code

https://github.com/strapi/strapi/pull/19606

**Pull request closed without merging**